evcache-core/src/main/java/com/netflix/evcache/EVCacheImpl.java

```diff
@@ -3259,7 +3260,21 @@ public <T> boolean add(String key, T value, Transcoder<T> tc, int timeToLive) th
     @Override
     public <T> EVCacheLatch add(String key, T value, Transcoder<T> tc, int timeToLive, Policy policy) throws EVCacheException {
         EVCacheClient[] clients = _pool.getEVCacheClientForWrite();
-        return this.add(key, value, tc, timeToLive, policy, clients, clients.length - _pool.getWriteOnlyEVCacheClients().length);
+        EVCacheClient[] writeOnlyClients = _pool.getWriteOnlyEVCacheClients();
+        // In case of adds , we skip adds to the pool if value is already present in the 1st client
```

Comment:
Any concerns with enabling the same behavior as SET calls? Fan out writes to all ASGs that can take writes.